### ISSUE TITLE
CA2016 doesn't work with `CancellationToken?` parameter

### ISSUE THREAD

**Issue by Vannevelj** [dotnet/roslyn-analyzers#6496]
### Analyzer

**Diagnostic ID**: [CA2016](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2016): `Forward the CancellationToken parameter to methods that take one`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: SDK 7.0.200-preview.22628.1

### Describe the bug

The analyzer does not trigger for methods that use `CancellationToken? token` or `CancellationToken? token = default`. It _does_ work for `CancellationToken token = default`.

### Steps To Reproduce

```cs
async Task DoThing(CancellationToken token) => await Other();

Task Other(CancellationToken? token = default) => Task.CompletedTask;
```

### Expected behavior

> CA2016	Forward the 'token' parameter to the 'Other' method or pass in 'CancellationToken.None' explicitly to indicate intentionally not propagating the token

### Actual behavior

No diagnostic is triggered

**Comment by OptimusPi** [comment]
Still an annoying issue
